cmd/sops/common/common.go

```diff
@@ -139,19 +170,40 @@ func IsIniFile(path string) bool {
 	return strings.HasSuffix(path, ".ini")
 }
 
+// DefaultStoreForFormat returns the correct format-specific implementation
+// of the Store interface given the format string
+func DefaultStoreForFormat(format string) Store {
+	storeConst, found := StoreConstructors[format]
+	if !found {
+		storeConst = StoreConstructors["binary"] // default
+	}
+	return storeConst()
+}
+
 // DefaultStoreForPath returns the correct format-specific implementation
 // of the Store interface given the path to a file
 func DefaultStoreForPath(path string) Store {
+	storeConst := StoreConstructors["binary"] // default
 	if IsYAMLFile(path) {
-		return &yaml.Store{}
+		storeConst = StoreConstructors["yaml"]
 	} else if IsJSONFile(path) {
-		return &json.Store{}
+		storeConst = StoreConstructors["json"]
 	} else if IsEnvFile(path) {
-		return &dotenv.Store{}
+		storeConst = StoreConstructors["dotenv"]
 	} else if IsIniFile(path) {
-		return &ini.Store{}
+		storeConst = StoreConstructors["ini"]
 	}
-	return &json.BinaryStore{}
+	return storeConst()
+}
+
+// DefaultStoreForPathOrFormat returns the correct format-specific implementation
+// of the Store interface given the format if specified, or the path to a file
+func DefaultStoreForPathOrFormat(path, format string) Store {
```

Comment:
This shouldn't be needed anymore -- if a `Format` is provided, a store for that format should exist.

We could potentially add a lint to make sure this is the case for us, since the Go compiler won't let us prove it.

Reply:
this is used in 3 places, for cli `--input-type`, cli `--output-type`, and in `decrypt.File`.